File: src/apps/kipo-dashboard/src/auth/core/domain/value-objects/AuthProvider.ts
export const AUTH_PROVIDER = {
  email: 'email',
  phone: 'phone',
  google: 'google',
  apple: 'apple',
  facebook: 'facebook',
} as const

export type AuthProvider = keyof typeof AUTH_PROVIDER

export const SOCIAL_PROVIDERS: AuthProvider[] = ['google', 'apple', 'facebook']

export const isSocialProvider = (p: AuthProvider): boolean =>
  SOCIAL_PROVIDERS.includes(p)

File: src/apps/kipo-platform/auth/value_objects/auth_provider.py
from shared.exceptions import BusinessRuleViolation

_SUPPORTED = frozenset({"email", "phone", "google", "facebook"})


class AuthProvider(str):
    def __new__(cls, value: str) -> "AuthProvider":
        normalized = value.strip().lower() if value else ""
        if normalized not in _SUPPORTED:
            raise BusinessRuleViolation(
                f"'{value}' is not a supported auth provider. Choose from: {', '.join(sorted(_SUPPORTED))}."
            )
        return super().__new__(cls, normalized)
